plugins/webhooks/index.js

```diff
@@ -45,73 +114,37 @@ function stopJob(job) {
     const prRef = options.prRef;
     const pipeline = options.pipeline;
-    const scm = request.server.app.pipelineFactory.scm;
-    const scmDisplayName = scm.getDisplayName({ scmContext });
-    const userDisplayName = `${scmDisplayName}:${username}`;
 
     return pipeline.getConfiguration(prRef)
-        // get permutations(s) for "main" job
-        .then(config => Promise.all([config.jobs.main, pipeline.jobs]))
-        // create a new job
-        .then(([permutations, jobs]) => {
+        // get configuration for all jobs
+        .then(config => Promise.all([config.jobs, pipeline.jobs]))
+        .then(([jobsConfig, jobs]) => {
+            const hasRequires = Object.keys(jobs).some(jobName => jobs[jobName].requires);
             const jobNamesArray = jobs.map(j => j.name);
-            const jobIndex = jobNamesArray.indexOf(name);
+            let jobName = options.name;
+            let jobIndex;
 
-            // if PR job already exists, update
-            if (jobIndex > -1) {
-                const job = jobs[jobIndex];
+            // OLD WORKFLOW DESIGN
+            if (!hasRequires) {
+                jobIndex = jobNamesArray.indexOf(jobName);
 
-                job.permutations = permutations;
-                job.archived = false;
-
-                return job.update();
+                return createOrUpdatePRJob(
+                    options, request, jobs[jobIndex], jobName, jobsConfig.main);
             }
 
-            // if the PR is new, create a job
-            return jobFactory.create({ pipelineId, name, permutations });
-        })
-        // log stuff
-        .then((job) => {
-            const jobId = job.id;
-
-            request.log(['webhook', hookId, jobId], `${name} created`);
-            request.log([
-                'webhook',
-                hookId,
-                jobId,
-                pipelineId
-            ], `${userDisplayName} selected`);
-
-            // create an event
-            return eventFactory.create({
-                pipelineId,
-                type: 'pr',
-                workflow: [job.name],
-                username,
-                scmContext,
-                sha,
-                causeMessage: `${options.action} by ${userDisplayName}`
-            })
-                .then(event =>
-                    buildFactory.create(
-                        { jobId, sha, username, scmContext, eventId: event.id, prRef })
-                );
-        })
-        // create a build
-        .then(build =>
-            request.log(['webhook', hookId, build.jobId, build.id],
-                `${name} started ${build.number}`));
+            // NEW WORKFLOW DESIGN
+            const prJobs = jobs.filter(j => j.requires && j.requires.includes('~pr'));
+
+            return Promise.all(prJobs.map((j) => {
+                jobName = `${jobName}-${j.name}`;
+                jobIndex = jobNamesArray.indexOf(jobName);
+
+                return createOrUpdatePRJob(
```

Comment:
This will create multiple event for each one of '~pr' jobs. We don't want it right?